### PR TITLE
Add CLI flag for HTTP API token path (VC)

### PR DESCRIPTION
**Pull request by syjn99**
## Issue Addressed

#6563

## Proposed Changes

- Added `http-token-path` flag for users who want to set the token path explicity.
- Added documentation for CLI reference, etc.

## Additional Info

- Throw error when `http-token-path` is actually a directory.
- If parent directory doesn't exist, make parent directory.

### CODE REVIEW

File: validator_client/src/cli.rs [hunk 1]
```diff
@@ -79,6 +79,19 @@ pub fn cli_app() -> Command {
...
+            Arg::new("http-token-path")
+                .long("http-token-path")
+                .value_name("HTTP_TOKEN_PATH")
+                .help(
+                    "Path to file containing the HTTP API token for validator client authentication. \
+                    If not specified, defaults to {validators-dir}/api-token.txt. \
+                    The file will be created if it does not exist.",
+                )
+                .action(ArgAction::Set)
+                .conflicts_with("datadir")
```
Comment (michaelsproul) [inline validator_client/src/cli.rs]: This should not conflict with `datadir`.

It should have `requires("http")` instead

File: validator_client/src/config.rs [hunk 2]
```diff
@@ -33,6 +36,8 @@ pub struct Config {
     pub validator_dir: PathBuf,
     /// The directory containing the passwords to unlock validator keystores.
     pub secrets_dir: PathBuf,
+    /// The path to the HTTP API token file for validator client authentication.
+    pub http_token_path: PathBuf,
```
Comment (michaelsproul) [inline validator_client/src/config.rs]: I think this field should be inside `validator_http_api::Config`

**Review by michaelsproul — CHANGES_REQUESTED** [review]
Thanks for the pull request!

This looks good on the whole, just a few minor tweaks and then we can merge

**Comment by syjn99** [comment]
@michaelsproul 
Fixed things you mentioned! Sorry for delay, enjoyed Devcon a lot 🙇‍♂️

File: book/src/help_vc.md [hunk 3]
```diff
@@ -69,6 +69,11 @@ Options:
           this server (e.g., http://localhost:5062).
       --http-port <PORT>
           Set the listen TCP port for the RESTful HTTP API server.
+      --http-token-path <HTTP_TOKEN_PATH>
+          Path to file containing the HTTP API token for validator client
+          authentication. If not specified, defaults to
+          {validators-dir}/api-token.txt. The file will be created if it does
+          not exist.
```
Comment (chong-he) [inline book/src/help_vc.md]: Small nitpick, The last sentence is a bit of ambiguous, I think the description is clear enough without the last sentence, hence suggesting to remove it.
```suggestion
          {validators-dir}/api-token.txt. 
```

**Review by chong-he — COMMENTED** [review]
Did some manual testing and it works well!

File: book/src/help_vc.md (hunk 3 above)
Comment (chong-he) [reply book/src/help_vc.md]: Edit: Need to change in the `cli.rs` too

**Comment by syjn99** [comment]
@michaelsproul @chong-he 
Could you please check this one more time? I applied reviews, fixed things that broke CI jobs.

**Comment by syjn99** [comment]
@michaelsproul It seems that there are some flaky tests for `make test-network`. I'm not sure but it might be related to #6646.

**Review by jimmygchen — APPROVED** [review]
Looks good to me, thanks for the PR!

**Review by michaelsproul — APPROVED** [review]

**Comment by jimmygchen** [comment]
@mergify queue

**Comment by michaelsproul** [comment]
@mergify dequeue

**Comment by jimmygchen** [comment]
@mergify requeue

**Pull request merged**